app/controllers/session_controller.rb

```diff
@@ -0,0 +1,45 @@
+
+class SessionController < ApplicationController
+
+  def authenticate
+    config = OpenIDConnect::Discovery::Provider::Config.discover! "https://did.app"
+
+    client = OpenIDConnect::Client.new({
+      identifier: ENV["CLIENT_ID"],
+      secret: ENV["CLIENT_SECRET"],
+      redirect_uri: "http://localhost:3000/session/callback",
```

Comment:
How do we get the host url so that the redirect is dynamic

Reply:
I think we may need to set `default_url_options`:

https://edgeguides.rubyonrails.org/action_controller_overview.html#default-url-options
https://www.benpickles.com/articles/70-dynamically-setting-rails-default_url_options-in-heroku-review-apps.

And then either interpolate in the host (and port) or make a route in `config/routes.rb`, maybe something like:

```
get '/session/callback', to:  'session#callback', as: 'session_callback'
```

...and then refer to `session_callback_url`.